inst/templates/ad_advs.R

```diff
@@ -0,0 +1,432 @@
+  bind_rows(who_wt_for_ht_girls %>%
+    mutate(SEX = "F")) %>%
+  mutate(MEASURE = "HEIGHT") %>%
+  rename(HEIGHT_LENGTH = Height) %>%
+  bind_rows(who_wt_for_lgth_boys %>%
+    mutate(SEX = "M") %>%
+    bind_rows(who_wt_for_lgth_girls %>%
+      mutate(SEX = "F")) %>%
+    mutate(MEASURE = "LENGTH") %>%
+    rename(HEIGHT_LENGTH = Length))
+
+# Load source datasets ----
+
+# Use e.g. `haven::read_sas()` to read in .sas7bdat, or other suitable functions
+# as needed and assign to the variables below.
+# For illustration purposes read in admiral test data
+
+# Once available to the Pharmaverse, needs to uncomment below data calls
+# data("vs_peds")
+# data("adsl_peds")
+
+vs <- vs_peds
+adsl <- adsl_peds %>% select(-DOMAIN)
+
+# When SAS datasets are imported into R using haven::read_sas(), missing
+# character values from SAS appear as "" characters in R, instead of appearing
+# as NA values. Further details can be obtained via the following link:
+# https://pharmaverse.github.io/admiral/articles/admiral.html#handling-of-missing-values # nolint
+
+vs <- convert_blanks_to_na(vs)
+
+# Lookup tables ----
+# Assign PARAMCD, PARAM, and PARAMN
+param_lookup <- tibble::tribble(
+  ~VSTESTCD, ~PARAMCD, ~PARAM, ~PARAMN,
+  "WEIGHT", "WEIGHT", "Weight (kg)", 1,
+  "HEIGHT", "HEIGHT", "Height (cm)", 2,
+  "BMI", "BMI", "Body Mass Index(kg/m^2)", 3,
+  "HDCIRC", "HDCIRC", "Head Circumference (cm)", 4,
+  NA_character_, "WTASDS", "Weight-for-age z-score", 5,
+  NA_character_, "WTAPCTL", "Weight-for-age percentile", 6,
+  NA_character_, "BMISDS", "BMI-for-age z-score", 7,
+  NA_character_, "BMIPCTL", "BMI-for-age percentile", 8,
+  NA_character_, "HDCSDS", "HDC-for-age z-score", 9,
+  NA_character_, "HDCPCTL", "HDC-for-age percentile", 10,
+  NA_character_, "WGTHSDS", "Weight-for-length/height Z-Score", 11,
+  NA_character_, "WGTHPCTL", "Weight-for-length/height Percentile", 12
+)
+attr(param_lookup$VSTESTCD, "label") <- "Vital Signs Test Short Name"
+
+# Get list of ADSL vars required for derivations
+adsl_vars <- exprs(SEX, BRTHDTC, TRTSDT, TRTEDT, TRT01A, TRT01P)
+
+advs <- vs %>%
+  # Join ADSL with VS (need BRTHDT for AAGECUR derivation)
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = adsl_vars,
+    by_vars = exprs(!!!get_admiral_option("subject_keys"))
+  ) %>%
+  mutate(BRTHDT = convert_dtc_to_dt(
+    BRTHDTC,
+    highest_imputation = "n",
+    date_imputation = "first",
+    min_dates = NULL,
+    max_dates = NULL,
+    preserve = FALSE
+  )) %>%
+  ## Calculate ADT, ADY ----
+  derive_vars_dt(
+    new_vars_prefix = "A",
+    dtc = VSDTC
+  ) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT)) %>%
+  ## Calculate Current Analysis Age AAGECUR ----
+  derive_vars_duration(
+    new_var = AAGECUR,
+    new_var_unit = AAGECURU,
+    start_date = BRTHDT,
+    end_date = ADT,
+    out_unit = "DAYS",
+    trunc_out = FALSE
+  )
+
+advs <- advs %>%
+  ## Add PARAMCD only - add PARAM etc later ----
```

Comment:
Is there any reason why PARAM is not added at the same time with PARAMCD?

Reply:
Nvm, I see PARAM being added after the z-scores and percentiles being derived in line 335, which do not have VSTESTCD.